src/Components/Nav/Nav.js

```diff
@@ -1,7 +1,108 @@
 import React from 'react';
+import styled from 'styled-components';
+import theme from '../../styles/Theme';
+import { Link } from 'react-router-dom';
+import { useEffect, useState } from 'react';
 
 const Nav = () => {
-  return <div>Nav</div>;
+  const [scrollY, setScrollY] = useState(0);
+
+  function handleScroll() {
+    setScrollY(window.pageYOffset);
+  }
+
+  useEffect(() => {
+    function scrollListener() {
+      window.addEventListener('scroll', handleScroll);
+    }
+
+    scrollListener();
+    // return () => {
+    //   window.removeEventListener('scroll', handleScroll);
+    // };
```

Comment:
함수 선언과 실행은 불필요한 것 같습니다.
그리고 unmount 시 eventListener 구독을 해제하는 로직(기존 주석처리)도 사용해주세요!

```suggestion
    window.addEventListener('scroll', handleScroll);
    
    return () => {
      window.removeEventListener('scroll', handleScroll);
    };
```

Reply:
그리고 useEffect에 빈 의존성 배열 추가해주시면 될 것 같습니다.

Reply:
gook !